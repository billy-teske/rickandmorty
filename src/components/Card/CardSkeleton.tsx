import styles from './CardSkeleton.module.css';
import listStyle from '../List/List.module.css';

const CardSkeleton = () => (
    <div className={listStyle.list}>
        {[1, 2, 3, 4, 5, 6, 7, 8].map((id) => (
            <div key={id} className={styles.skeleton} style={{
                animationDelay: `${id / 2}s`
            }}></div>
        ))}
    </div>
);

export default CardSkeleton;
