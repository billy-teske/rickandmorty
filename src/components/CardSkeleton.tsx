import { useEffect, useState } from 'react';
import styles from './CardSkeleton.module.css';
import listStyle from './List.module.css';

const CardSkeleton = () => {
    const [arrayCount, setArrayCount] = useState([0]);

    useEffect(() => {
        const intervalId = setInterval(() => {
            setArrayCount((prevCount) => [prevCount.length, ...prevCount]);
        }, 700);
        return () => clearInterval(intervalId);
    }, []);

    return (
        <div className={listStyle.list}>
            {arrayCount.map((id) => (
                <div key={id} className={styles.skeleton}></div>
            ))}
        </div>
    );
};

export default CardSkeleton;
