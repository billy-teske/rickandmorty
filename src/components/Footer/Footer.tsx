import styles from './Footer.module.css';

const Footer = () => (
    <footer className={styles.footer}>
        <span>Powered by Billy Teske</span>
    </footer>
);

export default Footer;
