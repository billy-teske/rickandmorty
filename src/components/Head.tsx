import Search from './Search';
import styles from './Head.module.css';

const Head = () => (
    <>
        <h1 className={styles.h1}>Rick and Morty</h1>
        <Search />
    </>
);

export default Head;
