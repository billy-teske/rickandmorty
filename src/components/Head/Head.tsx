import styles from './Head.module.css';

const Head = () => <h1 className={styles.h1}>Rick and Morty</h1>;

export default Head;
