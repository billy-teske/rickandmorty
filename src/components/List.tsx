import Card from './Card';
import Styles from './List.module.css';

const items = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14];

const List = () => (
    <ul className={Styles.list}>
        {items.map(item => (
            <Card
                key={item}
                name="Toxic Rick"
                status="Dead"
                species="Humanoid"
                type="Rick's Toxic Side"
                gender="Male"
                image="https://rickandmortyapi.com/api/character/avatar/368.jpeg"
            />
        ))}
    </ul>
);

export default List;
