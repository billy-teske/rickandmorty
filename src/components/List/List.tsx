import { TCharacter } from '../../api/getCharacter';
import Card from '../Card/Card';
import styles from './List.module.css';

interface IPropsList {
    characters: TCharacter[];
}

const List = ({ characters }: IPropsList) => (
    <ul className={styles.list}>
        {characters.map(character => (
            <Card
                key={character.id}
                name={character.name}
                status={character.status}
                species={character.species}
                type={character.type}
                gender={character.gender}
                image={character.image}
            />
        ))}
    </ul>
);

export default List;
