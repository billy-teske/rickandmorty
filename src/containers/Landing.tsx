import Head from '../components/Head/Head';
import List from '../components/List/List';
import Footer from '../components/Footer/Footer';
import CardSkeleton from '../components/Card/CardSkeleton';
import useCharacter from '../hooks/useCharacter';

const Landing = () => {
    const { isFetching, characters, error } = useCharacter();

    if (isFetching) {
        return (
            <>
                <Head />
                <CardSkeleton />
                <Footer />
            </>
        );
    }

    if (error) {
        return <span>error</span>;
    }

    return (
        <>
            <Head />
            <List characters={characters} />
            <Footer />
        </>
    );
};

export default Landing;