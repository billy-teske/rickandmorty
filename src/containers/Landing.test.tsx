import { render } from '@testing-library/react';
import Landing from './Landing';
import Head from '../components/Head/Head';
import List from '../components/List/List';
import Footer from '../components/Footer/Footer';
import CardSkeleton from '../components/Card/CardSkeleton';
import useCharacter from '../hooks/useCharacter';
import characterMock from '../api/__mock__/characterMock';

jest.mock('../components/Head/Head', () => jest.fn());
jest.mock('../components/List/List', () => jest.fn());
jest.mock('../components/Footer/Footer', () => jest.fn());
jest.mock('../components/Card/CardSkeleton', () => jest.fn());
jest.mock('../hooks/useCharacter', () => jest.fn());

(Head as jest.Mock).mockImplementation(() => <span>Head</span>);
(List as jest.Mock).mockImplementation(() => <span>List</span>);
(Footer as jest.Mock).mockImplementation(() => <span>Footer</span>);
(CardSkeleton as jest.Mock).mockImplementation(() => <span>CardSkeleton</span>);

describe('Landing Component', () => {
    it('should render Landing when load characters', () => {
        (useCharacter as jest.Mock).mockReturnValue({
            isFetching: true,
        });

        const result = render(
            <Landing />
        );

        expect(result).toMatchSnapshot();
    });

    it('should render Landing when the api have a error', () => {
        (useCharacter as jest.Mock).mockReturnValue({
            error: true,
        });

        const result = render(
            <Landing />
        );

        expect(result).toMatchSnapshot();
    });

    it('should render Landing with characters', () => {
        (useCharacter as jest.Mock).mockReturnValue({
            characters: characterMock.results
        });

        const result = render(
            <Landing />
        );

        expect(result).toMatchSnapshot();
    });
});
