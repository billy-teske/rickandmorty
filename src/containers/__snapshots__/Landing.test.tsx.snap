// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component should render Landing when load characters 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        Head
      </span>
      <span>
        CardSkeleton
      </span>
      <span>
        Footer
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      Head
    </span>
    <span>
      CardSkeleton
    </span>
    <span>
      Footer
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Landing Component should render Landing when the api have a error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        error
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      error
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Landing Component should render Landing with characters 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        Head
      </span>
      <span>
        List
      </span>
      <span>
        Footer
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      Head
    </span>
    <span>
      List
    </span>
    <span>
      Footer
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
